### PR TITLE
Introduce `ChunkStore::drop_entity_path`

### PR DESCRIPTION
**Pull request by teh-cmc**
Introduces a new API to drop an entity path in its entirety, unconditionally.

- Part of https://github.com/rerun-io/rerun/issues/6552
- Part of https://github.com/rerun-io/rerun/issues/1329

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6588?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6588?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6588)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_chunk_store/src/writes.rs [hunk 1]
```diff
@@ -161,4 +162,107 @@ impl ChunkStore {
...
+                    } = temporal_chunk_ids_per_time;
+
+                    per_start_time
+                        .into_values()
+                        .flat_map(|chunk_ids| chunk_ids.into_iter())
+                        .chain(
+                            per_end_time
+                                .into_values()
+                                .flat_map(|chunk_ids| chunk_ids.into_iter()),
+                        )
```
Comment (jleibs) [inline crates/re_chunk_store/src/writes.rs:230]: No harm in being thorough, but shouldn't every chunk_id occur both in `per_start_time` and `per_end_time`? Do we actually need to iterate both of them?

**Review by jleibs — APPROVED** [review]

**Comment by teh-cmc** [comment]
Reminder to get rid of the `do_not_protect` mechanism -- we definitely want to protect clears now... if anything we want to protect them even more than the rest.

File: crates/re_chunk_store/src/writes.rs (hunk 1 above)
Comment (teh-cmc) [reply crates/re_chunk_store/src/writes.rs:230]: Nah we don't, I went a bit overboard

**Comment by teh-cmc** [comment]
@rerun-bot full-check

**Pull request merged**